components/TransactionItem.js

```diff
@@ -0,0 +1,64 @@
+import styled from "styled-components";
+import { isToday, isYesterday, format } from "date-fns";
+
+export default function TransactionItem({ transaction }) {
+  const formatDate = (date) => {
+    if (!date) return "";
+    const newDateFormat = new Date(date);
+
+    if (isToday(newDateFormat)) {
+      return "Today";
+    } else if (isYesterday(newDateFormat)) {
+      return "Yesterday";
+    } else {
+      return format(newDateFormat, "dd.MM.yyyy");
+    }
+  };
+
+  const formatNumber = new Intl.NumberFormat("de-DE", {
+    style: "currency",
+    currency: "EUR",
+  }).format(transaction.amount);
+
+  return (
+    <>
+      <StyledDate>{formatDate(transaction.date)}</StyledDate>
+      <StyledCard>
+        <StyledId>{transaction.id}</StyledId>
+        <StyledCategory>{transaction.category}</StyledCategory>
+        <StyledAmount type={transaction.type}>{formatNumber}</StyledAmount>
+      </StyledCard>
+    </>
```

Comment:
by this strategy, the date is displayed next to every transaction. As a user I would expect to see all expenses from one day to be grouped and that grouping would have one date display. You can think about adapting your App in a later stage this way.

Reply:
Thank you for the hint. We also have planned it for later. We have added this todo to our project board to integrate to a later user story.